### PR TITLE
Factorio: use resources data

### PR DESCRIPTION
**Pull request by Berserker66**

### CODE REVIEW

File: worlds/factorio/Technologies.py [hunk 1]
```diff
@@ -203,18 +207,18 @@ def __init__(self, name, categories):
...
+    raw_recipes[f"mining-{resource_name}"] = current_recipe = {
+        "ingredients": {},
+        "products": {},
+        "energy": resource_data["mining_time"],
+        "category": resource_data["category"]
+    }
+    for name, data in resource_data["products"].items():
+        current_recipe["products"].update({data["name"]: data["amount"]})
+    if "required_fluid" in resource_data:
+        current_recipe["ingredients"] = {resource_data["required_fluid"]: resource_data["fluid_amount"]}
```
Comment (black-sliver) [inline worlds/factorio/Technologies.py:220]:
```suggestion
    raw_recipes[f"mining-{resource_name}"] = current_recipe = {
        "ingredients": {resource_data["required_fluid"]: resource_data["fluid_amount"]}
        if "required_fluid" in resource_data else {},
        "products": {data["name"]: data["amount"] for data in resource_data["products"].values()},
        "energy": resource_data["mining_time"],
        "category": resource_data["category"]
    }
```

Comment (Berserker66) [reply worlds/factorio/Technologies.py:220]: sure, though then we don't need to bind to current_recipe either anymore.

Comment (black-sliver) [reply worlds/factorio/Technologies.py:220]: oh, yeah, missed that. changes look good otherwise, i think

File: worlds/factorio/Technologies.py [hunk 2]
```diff
@@ -332,24 +336,7 @@ def get_rocket_requirements(silo_recipe: Recipe, part_recipe: Recipe, satellite_
...
-    Options.MaxSciencePack.option_chemical_science_pack:
-        {"advanced-circuit": 10, "steel-plate": 10, "solid-fuel": 10},
-    Options.MaxSciencePack.option_military_science_pack:
-        {"defender-capsule": 10, "stone-wall": 10, "coal": 10},
-    Options.MaxSciencePack.option_logistic_science_pack:
-        {"electronic-circuit": 10, "stone-brick": 10, "coal": 10},
-    Options.MaxSciencePack.option_automation_science_pack:
-        {"copper-cable": 10, "iron-plate": 10, "wood": 10}
-}
+free_sample_exclusions: Set[str] = all_ingredient_names | {"rocket-part"}
```
Comment (Ijwu) [inline worlds/factorio/Technologies.py:339]: 👍 Good rename

**Review by Ijwu — APPROVED** [review]

**Pull request merged**